QueryBuilder/Common/Helpers/LinqExpressionParser.cs

```diff
@@ -0,0 +1,315 @@
+                return $"{parentPropName}.{propName}";
+            }
+
+            // single level property
+            PropertyType = propInfo.PropertyType;
+            return propName;
+        }
+
+        private void ConvertIntToEnumIfNeeded()
+        {
+            if (IsEnumDeepCheck(PropertyType, Value, out var enumType))
+            {
+                Value = Enum.GetName(enumType, Value);
+            }
+        }
+
+        private static bool IsEnumDeepCheck(Type type, object value, out Type enumType)
+        {
+            enumType = null;
+            if (type == null || type.BaseType == null)
+            {
+                return false;
+            }
+
+            var underlyingType = Nullable.GetUnderlyingType(type);
+            if (underlyingType?.BaseType == typeof(Enum) && Enum.IsDefined(underlyingType, value))
+            {
+                enumType = underlyingType;
+                return true;
+            }
+
+            return false;
+        }
+
+        private static ComparisonOperators GetComparisonOperator(Expression expression)
+        {
+            return expression.NodeType switch
+            {
+                ExpressionType.Equal => ComparisonOperators.IsEqualTo,
+                ExpressionType.GreaterThan => ComparisonOperators.IsGreaterThan,
+                ExpressionType.GreaterThanOrEqual => ComparisonOperators.IsGreaterThanOrEqualTo,
+                ExpressionType.LessThan => ComparisonOperators.IsLessThan,
+                ExpressionType.LessThanOrEqual => ComparisonOperators.IsLessThanOrEqualTo,
+                ExpressionType.NotEqual => ComparisonOperators.NotEqualTo,
+                _ => throw new Exception($"Linq ExpressionType '{expression.NodeType}' is not supported by ADT.")
+            };
+        }
+
+        private static ScalarOperator GetScalarOperatorForMethod(string methodName)
+        {
+            return methodName switch
+            {
+                "StartsWith" => ScalarOperators.STARTSWITH,
+                "Contains" => ScalarOperators.CONTAINS,
+                "EndsWith" => ScalarOperators.ENDSWITH,
+                _ => throw new Exception($"Method '{methodName}' is not supported for conversion into a ScalarBinaryOperator")
+            };
+        }
+
+        private static ScalarUnaryOperator GetScalarOperatorForType(string typeName)
+        {
+            return typeName switch
+            {
+                "String" => ScalarOperators.IS_STRING,
+                "Boolean" => ScalarOperators.IS_BOOL,
+                "Double" => ScalarOperators.IS_NUMBER,
+                "Int16" => ScalarOperators.IS_NUMBER,
+                "Int32" => ScalarOperators.IS_NUMBER,
+                "Int64" => ScalarOperators.IS_NUMBER,
+                "Single" => ScalarOperators.IS_NUMBER,
+                "Object" => ScalarOperators.IS_OBJECT,
+                _ => throw new Exception($"Type '{typeName}' is not supported for conversion into a ScalarUnaryOperator")
```

Comment:
Unsure if a developer is going to know what to do with this exception message, mainly because they'd need to know what a `ScalarUnaryOperator` is (ADT refers to these as "type checking functions" in their [docs](https://learn.microsoft.com/en-us/azure/digital-twins/reference-query-functions).)